scipy/spatial/distance.py

```diff
@@ -1837,7 +1835,7 @@ def _cosine_cdist(XA, XB, dm):
     dm += 1
 
 
-def cdist(XA, XB, metric='euclidean', p=2, V=None, VI=None, w=None):
+def cdist(XA, XB, metric='euclidean', p=None, w=None, V=None, VI=None):
```

Comment:
It's good practice, but many people don't. Especially those coming from a language like Matlab, that doesn't have keyword args and where specifying >5 positional args is quite normal.

So while I'd like to see such things cleaned up, we just have a hard rule not to do that.